src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,110 @@
-export const Pagination = () => {};
+import { FC } from 'react';
+import cn from 'classnames';
+
+interface Props {
+  total: number;
+  perPage: number
+  currentPage: number
+  onPageChange: (pageId: number) => void;
+}
+
+export const Pagination: FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const natigationItems
+    = Array.from(
+      { length: Math.ceil(total / perPage) },
+      (_, i) => i + 1,
+    )
+      .map(item => (
+        <li
+          key={item}
+          className={cn(
+            'page-item',
+            { active: item === currentPage },
+          )}
+        >
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${item}`}
+            onClick={() => onPageChange(item)}
+          >
+            {item}
+          </a>
+        </li>
+      ));
+
+  const navListLength = natigationItems.length;
+
+  return (
+    <>
+      <ul className="pagination">
+        <li className={cn(
+          'page-item',
+          { disabled: currentPage <= 1 },
+        )}
+        >
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={currentPage <= 1}
+            onClick={() => {
+              if (currentPage > 1) {
+                onPageChange(currentPage - 1);
+              }
+            }}
+          >
+            «
+          </a>
+        </li>
+        {natigationItems}
+        <li className={cn(
+          'page-item',
+          { disabled: currentPage >= navListLength },
+        )}
+        >
+          <a
+            data-cy="nextLink"
+            className="page-link"
+            href="#next"
+            aria-disabled={currentPage >= navListLength}
```

Comment:
you use the same expression 2 times, so better create a variable

```suggestion
          { disabled: isTheNextButtonDisabled},
        )}
        >
          <a
            data-cy="nextLink"
            className="page-link"
            href="#next"
            aria-disabled={isTheNextButtonDisabled}
```